### PR TITLE
Offloads reminding user to AWS Lambda

### PR DESCRIPTION
**Pull request by bubakazouba**

### CODE REVIEW

File: deployremindinglambda.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    echo "provide the AWS Lambda function name (i.e ./deployremindinglambda.sh remindingLambdaDev remindingLambdaBucket)"
+    exit 1
+fi
+if [[ $s3BucketName == "" ]]; then
+    echo "provide the AWS S3 Bucket Name (i.e ./deployremindinglambda.sh remindingLambdaDev remindingLambdaBucket)"
+    exit 1
+fi
+
+cd remindinglambda
+npm install
```
Comment (arthurshir) [inline deployremindinglambda.sh:14]: What do you think about copying the config.json to this directory and referencing it in the lambda code? That way we don't have to send the config from the bot to AWS

Comment (arthurshir) [reply deployremindinglambda.sh:14]: Something like `cp ../config.json .` But maybe this breaks an encapsulation paradigm, what do you think?

File: utils/stringUtils.js [hunk 2]
```diff
@@ -1,3 +1,12 @@
+/**
+ * Encodes characters to safely display text to the user in HTML format (i.e ctx.replyWithHTML)
+ * @param  {String} text - text to be encoded
+ * @return {[String]} encodedText
```
Comment (arthurshir) [inline utils/stringUtils.js:4]: Should be `@return {String} encodedText`

File: remindinglambda/package.json [hunk 3]
```diff
@@ -0,0 +1,14 @@
+{
+  "name": "remindinglambda",
+  "version": "1.0.0",
+  "description": "",
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
```
Comment (arthurshir) [inline remindinglambda/package.json:7]: Nice, we should add a task to `NONFEATURETODOS.java` for adding unit tests to lambda

File: .gitignore [hunk 4]
```diff
@@ -8,3 +8,5 @@ updates.txt
 sftp-config.json
 .DS_STORE
 research-design/didnt test no to.java
+aws-config.json
+lambdaFunc.zip
```
Comment (arthurshir) [inline .gitignore:12]: Might want to ignore all zip files partly bc the name `lambdaFunc` isn't mentioned in the readme

File: deployremindinglambda.sh (hunk 1 above)
Comment (bubakazouba) [reply deployremindinglambda.sh:14]: My main concern around this is the fact that it's just something else for the dev to worry about(like you need to make sure your config at the time of deploying your lambda is prod ready). And something as simple as updating the config file would require a new deployment of the lambda, when it should just be updating 1 file

Comment (bubakazouba) [reply deployremindinglambda.sh:14]: confirmed that aws-sdk uses HTTPS as the default connection.

Comment (arthurshir) [reply deployremindinglambda.sh:14]: Cool, taking in account our current scale and workflow I am alright with us moving forward with this for now. What do you think about adding a todo to `NONFEATURETODOS.java` for designing a more secure approach of passing along configs to the lambda?.

**Review by arthurshir — APPROVED** [review]
Looks good, just added optional comment to add todo.

File: deployremindinglambda.sh (hunk 1 above)
Comment (bubakazouba) [reply deployremindinglambda.sh:14]: sounds good adding a github issue

Comment (bubakazouba) [reply deployremindinglambda.sh:14]: https://github.com/bubakazouba/remindmebot/issues/47

**Pull request merged**